app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt

```diff
@@ -65,12 +69,46 @@ class ProfileListAdapter(
             binding.TxtImg.setText(item.name[0].toString())
             binding.TxtImg.avatarBackgroundColor = bgColors[item.colorIndex]
             binding.timeStamp.text = item.timeInstance
+            binding.pauseSwitch.isChecked = item.pauseSwitch
 
             binding.profileCard.setOnClickListener {
                 val args = Bundle()
                 args.putParcelable("Profile", item)
                 val dialog = DetailsFragment.newInstance(args)
-                dialog.show((parentView.context as FragmentActivity).supportFragmentManager, "DialogFragment")
+                dialog.show(
+                    (parentView.context as FragmentActivity).supportFragmentManager,
+                    "DialogFragment"
+                )
+            }
+            binding.pauseSwitch.setOnClickListener {
+                if (!binding.pauseSwitch.isChecked) {
+                    item.pauseSwitch = false
+                    WorkManager.getInstance(profileViewModel.getApplication())
+                        .cancelAllWorkByTag(item.profileId.toString())
+                    if (StoreSession.readLong(AppConstants.ACTIVE_PROFILE_ID) == item.profileId)
+                        audioManager.ringerMode = StoreSession.readInt(AppConstants.RINGTONE_MODE)
+                } else {
+                    var smin: Int = 0
+                    var shr: Int = 0
+                    item.pauseSwitch = true
+                    val current: Calendar = Calendar.getInstance()
+                    if (StoreSession.readLong(AppConstants.ACTIVE_PROFILE_ID) == item.profileId) {
+                        if (current.get(Calendar.MINUTE) == 59) {
+                            smin = 0
+                            if (current.get(Calendar.HOUR_OF_DAY) == 23)
+                                shr = 0
+                            else
+                                shr = current.get(Calendar.HOUR_OF_DAY) + 1
+                        } else {
+                            smin = current.get(Calendar.MINUTE) + 1
+                            shr = current.get(Calendar.HOUR_OF_DAY)
+                        }
+                        WorkManagerHelper.setAlarms(item, shr, smin)
+                    } else {
+                        WorkManagerHelper.setAlarms(item)
```

Comment:
@rob729 In the earlier code there was a small problem - If an active profile was paused and resumed, it's starting time was required to rescheduled for stating the work again. But earlier the start time was getting changed in the data base also, which was being reflected in Details Fragment also.

Now that has been fixed. I have used these lines so that the rescheduling start time gets changed only for restarting the work and not in the database.

Reply:
I have tested the app after making these changes, it is working fine. Please check once and if there's any problem please let me know.